### PR TITLE
feature/56478 2nd PR for ITHC setup

### PR DESCRIPTION
**Pull request by GuyHarwood**
- update seed scripts for SM and TD roles to nullify school reference
- enhance shell scripts to renew keys with optional auto key vault updates
- create YAML build task for renewing azure keys with configurable UI parameters

### CODE REVIEW

File: deploy/keys/functions.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+# https://learn.microsoft.com/en-us/cli/azure/functionapp/keys?view=azure-cli-latest#az-functionapp-keys-set
+az functionapp keys set -g $RES_GROUP -n $FUNCTION_APP_NAME --key-type masterKey
+
+# skip key vault update if requested
+if [ $UPDATE_KV_SECRET == "False" ]; then exit 0; fi
+echo "UPDATE_KV_SECRET set to $UPDATE_KV_SECRET. Updating key vault secret..."
+
+## update key vault value...
+az keyvault secret set --vault-name $KEY_VAULT_NAME --name "TODO" --value $KEY_VALUE  -o none
```
Comment (jon-shipley) [inline deploy/keys/functions.sh:29]: Why is the name TODO?

File: deploy/keys/keyvault-update.sh [hunk 2]
```diff
@@ -5,6 +5,9 @@
 # -----------
 # Service Bus auth rule names are hardcoded
 
+# should not be used as limited by only supporting the primary keys
+exit -1
```
Comment (jon-shipley) [inline deploy/keys/keyvault-update.sh:9]: This should probably also print a warning to stderr

**Review by jon-shipley — APPROVED** [review]
LGTM

Just one suggestion and one question.

File: deploy/keys/functions.sh (hunk 1 above)
Comment (GuyHarwood) [reply deploy/keys/functions.sh:29]: on line 10 the script exits automatically and warns the user.  the comments above it prove context on the issue and link to an article about it.  its kept only for reference

File: deploy/keys/keyvault-update.sh (hunk 2 above)
Comment (GuyHarwood) [reply deploy/keys/keyvault-update.sh:9]: done

**Pull request merged**